src/Moderation/Factory/Report.php

```diff
@@ -22,58 +22,126 @@
-			$row['uid'],
+			$posts ?? new Collection\Report\Posts(),
+			$rules ?? new Collection\Report\Rules(),
+			$row['public-remarks'],
+			$row['private-remarks'],
+			$row['edited'] ? new \DateTimeImmutable($row['edited'], new \DateTimeZone('UTC')) : null,
+			$row['status'],
+			$row['resolution'],
+			$row['assigned-uid'],
+			$row['last-editor-uid'],
 			$row['id'],
 		);
 	}
 
 	/**
 	 * Creates a Report entity from a Mastodon API /reports request
 	 *
-	 * @see \Friendica\Module\Api\Mastodon\Reports::post()
-	 *
-	 * @param int    $uid
+	 * @param array  $rules      Line-number indexed node rules array, see System::getRules(true)
 	 * @param int    $reporterId
 	 * @param int    $cid
+	 * @param int    $gsid
 	 * @param string $comment
+	 * @param string $category
 	 * @param bool   $forward
 	 * @param array  $postUriIds
+	 * @param array  $ruleIds
+	 * @param ?int   $uid
 	 * @return Entity\Report
-	 * @throws \Exception
+	 * @see \Friendica\Module\Api\Mastodon\Reports::post()
 	 */
-	public function createFromReportsRequest(int $reporterId, int $cid, string $comment = '', string $category = null, string $rules = '', bool $forward = false, array $postUriIds = [], int $uid = null): Entity\Report
+	public function createFromReportsRequest(array $rules, int $reporterId, int $cid, int $gsid, string $comment = '', string $category = '', bool $forward = false, array $postUriIds = [], array $ruleIds = [], int $uid = null): Entity\Report
 	{
+		if (count($ruleIds)) {
+			$categoryId = Entity\Report::CATEGORY_VIOLATION;
+		} elseif ($category == 'spam') {
+			$categoryId = Entity\Report::CATEGORY_SPAM;
+		} else {
+			$categoryId = Entity\Report::CATEGORY_OTHER;
+		}
+
 		return new Entity\Report(
 			$reporterId,
 			$cid,
-			new \DateTime('now', new \DateTimeZone('UTC')),
+			$gsid,
+			$this->clock->now(),
+			$categoryId,
+			$uid,
 			$comment,
-			$category,
-			$rules,
 			$forward,
-			$postUriIds,
-			$uid,
+			new Collection\Report\Posts(array_map(function ($uriId) {
+				return new Entity\Report\Post($uriId);
+			}, $postUriIds)),
+			new Collection\Report\Rules(array_map(function ($lineId) use ($rules) {
+				return new Entity\Report\Rule($lineId, $rules[$lineId] ?? '');
+			}, $ruleIds)),
+		);
+	}
+
+	public function createFromForm(array $rules, int $cid, int $reporterId, int $categoryId, array $ruleIds, string $comment, array $uriIds): Entity\Report
+	{
+		$contact = Contact::getById($cid, ['gsid']);
+		if (!$contact) {
+			throw new \InvalidArgumentException('Contact with id: ' . $cid . ' not found');
+		}
+
+		if (!in_array($categoryId, Entity\Report::CATEGORIES)) {
+			throw new \OutOfBoundsException('Category with id: ' . $categoryId . ' not found in set: [' . implode(', ', Entity\Report::CATEGORIES) . ']');
+		}
+
+		return new Entity\Report(
+			Contact::getPublicIdByUserId($reporterId),
+			$cid,
+			$contact['gsid'],
+			$this->clock->now(),
+			$categoryId,
+			$reporterId,
+			$comment,
+			false, // Forward?
```

Comment:
Can you include the forward as well? Although we cannot transmit reports at the moment, it shouldn't be a problem doing so.